Thanks! This looks good - although I do see build errors for ESP32 boards now.

Specifically:

```
targets/esp32/jswrap_esp32.c: In function 'esp32IsValidDeepSleepWakeupPin':
targets/esp32/jswrap_esp32.c:82:21: error: implicit declaration of function 'pinToESP32Pin' [-Werror=implicit-function-declaration]
   gpio_num_t gpio = pinToESP32Pin(pin);
                     ^
targets/esp32/jswrap_esp32.c:84:10: error: implicit declaration of function 'esp_sleep_is_valid_wakeup_gpio' [-Werror=implicit-function-declaration]
   return esp_sleep_is_valid_wakeup_gpio(gpio);
          ^
targets/esp32/jswrap_esp32.c: In function 'jswrap_ESP32_lightSleep':
targets/esp32/jswrap_esp32.c:280:14: error: 'ESP_ERR_SLEEP_TOO_SHORT_SLEEP_DURATION' undeclared (first use in this function)
       err != ESP_ERR_SLEEP_TOO_SHORT_SLEEP_DURATION &&
              ^
targets/esp32/jswrap_esp32.c:280:14: note: each undeclared identifier is reported only once for each function it appears in
targets/esp32/jswrap_esp32.c:281:14: error: 'ESP_ERR_SLEEP_REJECT' undeclared (first use in this function)
       err != ESP_ERR_SLEEP_REJECT) {
```

The CI builds for ESP32C3/S3 fail too but the error log for those is long enough that Github freaks out while trying to display it :) Seems odd though as this was specifically for the C3?

Do you think there might be something that wasn't committed?

--- a/targets/esp32/jswrap_esp32.c
+++ b/targets/esp32/jswrap_esp32.c
@@ -35,10 +35,18 @@
 #include "BLE/esp32_bluetooth_utils.h"
 #endif
 #include "jshardwareESP32.h"
+#include "jshardwareSpi.h"
 
-#include "jsutils.h"
-#include "jsinteractive.h"
-#include "jsparse.h"
+#ifndef ESP_IDF_VERSION_MAJOR
+#define ESP_IDF_VERSION_MAJOR 3
+#endif
+
+#ifndef ESP_ERR_SLEEP_REJECT
+#define ESP_ERR_SLEEP_REJECT ESP_ERR_INVALID_STATE
+#endif
+#ifndef ESP_ERR_SLEEP_TOO_SHORT_SLEEP_DURATION
+#define ESP_ERR_SLEEP_TOO_SHORT_SLEEP_DURATION ESP_ERR_INVALID_ARG
+#endif
 
 /*JSON{
   "type": "class",
@@ -81,7 +89,20 @@ void jswrap_ESP32_reboot() {
 static bool esp32IsValidDeepSleepWakeupPin(Pin pin) {
   gpio_num_t gpio = pinToESP32Pin(pin);
   if (gpio < 0 || gpio >= GPIO_NUM_MAX) return false;
+#if ESP_IDF_VERSION_MAJOR >= 4
   return esp_sleep_is_valid_wakeup_gpio(gpio);
+#else
+  switch (gpio) {
+    case GPIO_NUM_0: case GPIO_NUM_2: case GPIO_NUM_4:
+    case GPIO_NUM_12: case GPIO_NUM_13: case GPIO_NUM_14: case GPIO_NUM_15:
+    case GPIO_NUM_25: case GPIO_NUM_26: case GPIO_NUM_27:
+    case GPIO_NUM_32: case GPIO_NUM_33: case GPIO_NUM_34: case GPIO_NUM_35:
+    case GPIO_NUM_36: case GPIO_NUM_37: case GPIO_NUM_39:
+      return true;
+    default:
+      return false;
+  }
+#endif
 }
 
 static uint64_t esp32DeepSleepGpioMask(Pin pin) {

